modules/core/src/main/java/org/apache/ignite/internal/processors/cache/transactions/TxSavepointLocal.java

```diff
@@ -0,0 +1,119 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.internal.processors.cache.transactions;
+
+import java.util.LinkedHashMap;
+import org.apache.ignite.internal.util.tostring.GridToStringInclude;
+import java.util.Collection;
+import java.util.Map;
+
+/**
+ * Transaction savepoint implementation for near local transactions.
+ */
+public class TxSavepointLocal implements TxSavepoint {
```

Comment:
perhaps, we don't need interface with the only implementation now. We could create it when we need it